### ISSUE TITLE
DeMorgan for `all`/`any` in attrs

### ISSUE THREAD

**Issue by llogiq** [rust-lang-nursery/rust-clippy#3176]
We should lint attributes of the style `#[cfg(all(not(_), not(_), ..))]` and `#[cfg(any(not(_), not(_), ..))]` and suggest `#[cfg(not(any(_, _, ..))]` and `#[cfg(not(all(_, _, ..))]`, respectively. Bonus points for further simplification.

**Comment by JayKickliter** [comment]
DeMorgan?

**Comment by llogiq** [comment]
Yes. and(not(x), not(y)) == not(or(x, y)) and vice versa. Just replace and with all and or with any.

**Comment by clarcharr** [comment]
Don't we have the opposite lint for boolean expressions? I.e. `!(x == y && z > w)` becomes `x != y || z <= w`. I think these two go well together.

**Comment by xFrednet** [comment]
Hey @goth-turtle, just a warning. Clippy has some problems dealing with cfg attributes, as they are not preserved as part of the AST/HIR. This behavior is actually defined in the rust reference (IDK why though). I'm pretty sure that there is currently no good way to implement this lint. At some point, we have to find a solution to deal with cfg attributes, as they also mess with some other lints. If you find a solution, feel free to create a PR, but otherwise it might be better to search for a different issue :sweat_smile:

**Comment by goth-turtle** [comment]
Ah, thanks for letting me know!
